File: 351001/Betenya/backend/src/auth/jwt.service.ts
import { Injectable } from '@nestjs/common';
import * as jwt from 'jsonwebtoken';

const SECRET = process.env.JWT_SECRET || 'distcomp-jwt-secret-key-2024';
const EXPIRES_IN = '1h';

export interface JwtPayload {
  sub: string;
  id: number;
  role: string;
  iat?: number;
  exp?: number;
}

@Injectable()
export class JwtService {
  sign(payload: { sub: string; id: number; role: string }): string {
    return jwt.sign(payload, SECRET, { expiresIn: EXPIRES_IN });
  }

  verify(token: string): JwtPayload {
    return jwt.verify(token, SECRET) as JwtPayload;
  }
}

File: 351003/Guzaev/publisher/app/security/jwt_service.py
from datetime import datetime, timedelta
from jose import JWTError, jwt
from passlib.context import CryptContext

SECRET_KEY = "distcomp-secret-key-2026"
ALGORITHM = "HS256"
ACCESS_TOKEN_EXPIRE_MINUTES = 60

pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")

def hash_password(password: str) -> str:
    return pwd_context.hash(password)

def verify_password(plain: str, hashed: str) -> bool:
    return pwd_context.verify(plain, hashed)

def create_access_token(login: str, role: str) -> str:
    now = datetime.utcnow()
    payload = {
        "sub": login,
        "role": role,
        "iat": now,
        "exp": now + timedelta(minutes=ACCESS_TOKEN_EXPIRE_MINUTES)
    }
    return jwt.encode(payload, SECRET_KEY, algorithm=ALGORITHM)

def decode_token(token: str) -> dict:
    try:
        return jwt.decode(token, SECRET_KEY, algorithms=[ALGORITHM])
    except JWTError:
        return None